packages/iov-core/src/multichainsigner.ts

```diff
@@ -28,19 +30,42 @@ async function connectChain(x: ChainConnector): Promise<Chain> {
   };
 }
 
+/**
+ * TransactionSigner is just the methods on `UserProfile` that we need in `MultiChainSigner`.
+ * By only requiring this interface, we allow the use of other implementations with custom
+ * logic for key derivation, etc.
+ */
+export interface Profile {
+  readonly signTransaction: (
+    id: WalletId,
+    identity: PublicIdentity,
```

Comment:
The existence of multiple wallets and the type `WalletId` is specific to the implementation. If we can guarantee that a `PublicIdentity` (soon a chainId-pubkey-pair) is unique within a Profile, the `id` parameter can be dropped. Such a check is probably not a bad idea.

However, out of scope for this one.

Reply:
I agree. Good idea but out of scope.

I would be happy to have `PublicIdentity` be globally unique over all `Wallets` and the `WalletId` disappear as unnecessary details.

But I think that is a different PR